string type
float64 theta
---
int8 done
